File: Python version/specializations/dlv2/DLV2AnswerSets.py
from languages.asp.AnswerSets import AnswerSets
from parsers.asp.ASPSolversParser import ASPSolversParser

class DLV2AnswerSets(AnswerSets):
    """Represents Dlv2's answersets"""
    
    def __init__(self, out, err=None):
        super(DLV2AnswerSets, self).__init__(out, err)
        
    def _parse(self):
        ASPSolversParser.parseDLV2(self, self._output, True)

File: Java version/app/src/main/java/it/unical/mat/embasp/specializations/dlv2/DLV2AnswerSets.java
package it.unical.mat.embasp.specializations.dlv2;

import it.unical.mat.embasp.languages.asp.AnswerSets;
import it.unical.mat.parsers.asp.ASPSolversParser;

/** represents Dlv2's answersets */

public class DLV2AnswerSets extends AnswerSets {

	public DLV2AnswerSets(final String answersets) {
		super(answersets);
	}

	public DLV2AnswerSets(final String out, final String err) {
		super(out, err);
	}

	@Override
	protected void parse() {
		ASPSolversParser.parseDLV2(this, output, true);
	}
}
